gap/attributes/attr.gi

```diff
@@ -1045,13 +1045,173 @@ function(S)
 
   return
     function(x, y)
-      local R;
-      R := RClass(S, x);
-      return IsGreensLessThanOrEqual(R, RClass(S, y))
-        and ForAny(Idempotents(R), e -> e * y = x);
+      local E;
+      E := Idempotents(S);
+      return ForAny(E, e -> e * y = x)
+        and ForAny(E, f -> y * f = x);
     end;
 end);
 
+SEMIGROUPS.ExistsTransversal := function(a, b, n)
+    local exists, class, i;
+    if DegreeOfTransformation(a) > n or
+        DegreeOfTransformation(b) > n then
+        ErrorNoReturn("Transformation does not",
+        "belong to a transformation semigroup of degree n");
+    fi;
+    for class in KernelOfTransformation(a, n) do
+        exists := false;
+        for i in [1 .. Size(class)] do
+            if class[i] ^ a = class[i] ^ b then
+                exists := true;
+                break;
+            fi;
+        od;
+        if not exists then
+            return false;
+        fi;
+    od;
+    return true;
+end;
+
+InstallMethod(RegularLeqTransformationSemigroupNC,
+#intended for internal use; see MitschLeqSemigroup for public use
+"for a transformation semigroup",
+[IsTransformationSemigroup],
+function(S)
+    local deg;
+    deg := DegreeOfTransformationSemigroup(S);
+    return
+        function(x, y)
+            # Only returns the correct answer if y is a regular element
+            return IsRefinementKernelOfTransformation(x, y, deg)
+             and SEMIGROUPS.ExistsTransversal(x, y, deg);
+        end;
+end);
+
+InstallMethod(MitschLeqSemigroupNC,
+"for a semigroup",
+[IsSemigroup],
+function(S)
+    return 
+        function(x, y)
+            if x = y then
+                return true;
+            else
+                return ForAny(Elements(S), s -> x = x * s and x = y * s) and
+                    ForAny(Elements(S), t -> t * y = x and t * x = x);
```

Comment:
Aren't semigroups iterable? Because if they are, can't you just write
```suggestion
                return ForAny(S, s -> x = x * s and x = y * s) and
                    ForAny(S, t -> t * y = x and t * x = x);
```